cluster/gce/util.sh

```diff
@@ -334,8 +334,8 @@ function detect-node-names() {
   detect-project
   INSTANCE_GROUPS=()
   INSTANCE_GROUPS+=($(gcloud compute instance-groups managed list \
-    --zones "${ZONE}" --project "${PROJECT}" \
-    --regexp "${NODE_INSTANCE_PREFIX}-.+" \
+    --project "${PROJECT}" \
+    --filter "name ~ '${NODE_INSTANCE_PREFIX}-.+' AND zone:(${ZONE})" \
```

Comment:
why is it sometimes `zone:(${ZONE})` and sometimes `zone=(${ZONE})`?

Reply:
ah, I guess `:(` matches patterns, whereas `=(` matches exactly, per https://cloud.google.com/sdk/gcloud/reference/topic/filters.

this is probably fine, though I'm guessing `=(` is maybe more correct?

Reply:
@ixdy the reasoning in full [is here](https://github.com/kubernetes/kubernetes/pull/50398#issuecomment-324380995) but `:` seems to be the way `gcloud` push as `=` flat out doesn't work for zones

Reply:
@ixdy Very sorry - I missed [one zone here](cluster/validate-cluster.sh) I'll fix that

Reply:
ok, there now shouldn't be any occurrences of `zone=` inside a filter